inst/d3-jetpack/d3-jetpack.js

```diff
@@ -0,0 +1,577 @@
+        if (!polygonInside(c, a, b)) {
+          subject.push(polygonIntersect(c, d, a, b));
+        }
+        subject.push(d);
+      } else if (polygonInside(c, a, b)) {
+        subject.push(polygonIntersect(c, d, a, b));
+      }
+      c = d;
+    }
+    if (closed) subject.push(subject[0]);
+    a = b;
+  }
+
+  return subject;
+};
+
+function polygonInside(p, a, b) {
+  return (b[0] - a[0]) * (p[1] - a[1]) < (b[1] - a[1]) * (p[0] - a[0]);
+}
+
+// Intersect two infinite lines cd and ab.
+function polygonIntersect(c, d, a, b) {
+  var x1 = c[0], x3 = a[0], x21 = d[0] - x1, x43 = b[0] - x3,
+      y1 = c[1], y3 = a[1], y21 = d[1] - y1, y43 = b[1] - y3,
+      ua = (x43 * (y1 - y3) - y43 * (x1 - x3)) / (y43 * x21 - x43 * y21);
+  return [x1 + ua * x21, y1 + ua * y21];
+}
+
+// Returns true if the polygon is closed.
+function polygonClosed(coordinates) {
+  var a = coordinates[0],
+      b = coordinates[coordinates.length - 1];
+  return !(a[0] - b[0] || a[1] - b[1]);
+}
+
+var prev = {};
+
+var timer$1 = function(fn, delay, time, name){
+  if (prev[name]) prev[name].stop();
+
+  var newTimer = d3Timer.timer(fn, delay, time, name);
+  if (name) prev[name] = newTimer;
+
+  return newTimer
+};
+
+var prev$1 = {};
+
+var interval$1 = function(fn, delay, time, name){
+  if (prev$1[name]) prev$1[name].stop();
+
+  var newTimer = d3Timer.interval(fn, delay, time, name);
+  if (name) prev$1[name] = newTimer;
+
+  return newTimer
+};
+
+var prev$2 = {};
+
+var timeout$1 = function(fn, delay, time, name){
+  if (prev$2[name]) prev$2[name].stop();
+
+  var newTimer = d3Timer.timeout(fn, delay, time, name);
+  if (name) prev$2[name] = newTimer;
+
+  return newTimer
+};
+
+d3Selection.selection.prototype.translate = translateSelection;
+d3Transition.transition.prototype.translate = translateSelection;
+d3Selection.selection.prototype.append = append;
+d3Selection.selection.prototype.insert = insert;
+d3Selection.selection.prototype.parent = parent;
+d3Selection.selection.prototype.selectAppend = selectAppend;
+d3Selection.selection.prototype.tspans = tspans;
+d3Selection.selection.prototype.appendMany = appendMany;
+d3Selection.selection.prototype.at = at;
+d3Selection.selection.prototype.st = st;
+d3Transition.transition.prototype.at = at;
+d3Transition.transition.prototype.st = st;
+d3Selection.selection.prototype.prop = d3Selection.selection.prototype.property;
+
+exports.wordwrap = wordwrap;
+exports.parseAttributes = parseAttributes;
+exports.f = f;
+exports.ascendingKey = ascendingKey;
+exports.descendingKey = descendingKey;
+exports.conventions = conventions;
+exports.drawAxis = drawAxis;
+exports.attachTooltip = attachTooltip;
+exports.loadData = loadData;
+exports.nestBy = nestBy;
+exports.round = round;
+exports.clamp = clamp;
+exports.polygonClip = polygonClip;
+exports.timer = timer$1;
+exports.interval = interval$1;
+exports.timeout = timeout$1;
+
+Object.defineProperty(exports, '__esModule', { value: true });
+
+}
+
+(function () {
+
```

Comment:
Don't we also need to run `init_d3_jetpack` for `window.d3`?

Reply:
I see your comment on `d3-jetpack requires d3 version 4 or higher`, did you find a case where jetpack breaks under D3v3 or docs that list D3v3 as unsupported?

Reply:
Yeah, I recall that when we tested with d3 v3 you got a runtime error (missing method).

If there is some hack or patch to achieve d3 compatibility we could always apply that later if/when there is user demand.